plugins/sectionManager.js

```diff
@@ -1,67 +1,54 @@
 const
-  fs = require('fs'),
   path = require('path'),
-  markdownIt = require('markdown-it'),
+  globby = require('globby').sync,
   config = require('../getConfig').get();
 
-const SECTION_REGEX = /(\[section=)[a-zA-Z0-9\-]+\]/g;
+const SECTION_REGEX = /\[section=([a-zA-Z0-9\-]+)\]/g;
 
-module.exports = {
+module.exports = function plugin () {
+  const process = function process (file, files) {
+    const content = files[file].contents.toString();
 
-  listSections: [],
+    let has_section = false;
+    let sections = [];
 
-  process(filename, data, handlebars) {
-    let fileString = data.contents.toString();
-    const match = fileString.match(SECTION_REGEX);
+    const out = content.replace(SECTION_REGEX, (match, name) => {
+      let replacement = '';
 
-    if (match) {
-      fileString = this.injectSection(match, filename, fileString, handlebars);
-    }
-
-    return {
-      has_section: match,
-      sections: this.listSections.join(),
-      fileContent: Buffer.from(fileString)
-    };
-  },
+      for (const fn of globby(path.join(__dirname, `../src/${path.dirname(file)}/${name}*`))) {
+        has_section = true;
+        sections.push(name);
 
-  injectSection(match, filename, fileString, handlebars) {
+        const lng = fn.replace(/^.*\.([^.]+)\.[^.]+$/, '$1');
```

Comment:
It's really nitpicking sorry but could you be more explicit in the variable name since its a regex result ? I have to read the regex to understand (and everybody hate to read regex :p )